### ISSUE TITLE
Migrate "Orders > Orders > Add new" page

### ISSUE THREAD

**Issue by matks** [PrestaShop/PrestaShop#10583]
Type: page
Part of Symfony migration project

## Status

Migration block per block

Todo:
- [x] Migrate Add action https://github.com/PrestaShop/PrestaShop/pull/12645
- [x] Migrate block "customer"
- [x] Migrate block "products" (a.k.a cart) https://github.com/PrestaShop/PrestaShop/pull/15814
- [x] Migrate block "cart rules" https://github.com/PrestaShop/PrestaShop/pull/15995
- [x] Migrate block "adresses"
- [x] Migrate block "shipping" https://github.com/PrestaShop/PrestaShop/pull/15899
- [x] Migrate "summary" block [WIP] https://github.com/PrestaShop/PrestaShop/pull/16207
- [x] Add "add an order" button to listing
- [x] Add a modale for "Details" of a customer #17090
- [x] Add a modale for "Add new voucher" #17090

Centralized Mail sendings: see https://github.com/PrestaShop/PrestaShop/pull/13554#discussion_r282534335

### Controller / template
- [ ] All controller actions are migrated and protected with `@AdminSecurity` annotations when eligible and `@DemoRestricted` annotations when eligible
- [ ] All links targeting this page are updated: either using _legacy_link or updating the code
- [ ] The new SF routes have `_legacy_link` and `_legacy_parameters` provided and explained
- [ ] Controller catches Domain Exception and translate them nicely
- [ ] Twig templates must contain relevant blocks to allow partial extension by modules
- [ ] JS being used on the page can be used as enabled extensions and use a PageMap for selectors
##### Conventions
- [ ] The new controller actions comply with our naming convention
- [ ] The new SF routes names and paths comply with our naming convention
- [ ] The new Twig template names comply actions with our naming convention
- [ ] Created files (controller, JS assets, CSS assets and twig views) must be in the right folder following the convention
- [ ] CSS being added must follow theme conventions

#### Core
- [ ] Commands and Queries only use native types but internally use VOs
- [ ] All Handlers must have an Interface
##### Conventions
- [ ] The created SF service names comply with our naming convention
- [ ] The created files must follow the folder convention (Domain, Core, Adapter...)

#### Polishing
- [ ] The legacy controller and templates files are gone
- [ ] If PR includes a new concept/mechanism, documentation must be written
- [ ] JS and CSS assets must be compiled and up-to-date
- [ ] License headers must be up-to-date
- [ ] New hooks have been added to the hooks list and the SQL updates
- [ ] Performance must be correct: page loading time, filtering/sorting/searching response time, actions response time

#### Testing
- [ ] Commands and Queries are tested with Behat tests
- [ ] Eligible Core classes are covered with unit tests*
- [ ] Controller is covered with survival test
- [ ] (soon) Controller is covered with E2E tests using puppeeter
- [ ] If we break E2E tests because IDs change, we must update the E2E tests

*Eligible = easy to isolate and unit tests provide values (for example testing getters/setters is useless)

**Comment by marionf** [comment]
**Features**
- [ ] #16131 (need to check the complexity)
- [ ] #16129

**Issues Must-Have sorted by priority high to low :**
- [ ] #11492
- [x] #12442
- [ ] #11115
- [x] #12521
- [x] #13809
- [x] #9850
- [ ] #13880

**Issues Should-Have sorted by priority high to low :**
- [ ] #9676 (need to wait the new design)
- [ ] #9783
- [ ] #10988
- [x] #9689
- [x] #13061
- [x] #14417

**Issues Nice to Have sorted by priority high to low :**
- [ ] #12746
- [ ] #9735
- [ ] #12682
- [x] #9876 
- [x] #9830 
- [ ] #12588
- [x] #11034
- [x] #11176
- [x] #13782
- [ ] #10393

**Comment by zuk3975** [comment]
@matks, something to discuss - after each action of order creation, legacy used to return some full object models to view (like carrier, invoice, product etc.) containing lot of unnecessary data. Regarding BC, modules might be using that data? How should it be handled? currently I am only returning what is needed for rendering.

**Comment by matks** [comment]
> @matks, something to discuss - after each action of order creation, legacy used to return some full object models to view (like carrier, invoice, product etc.) containing lot of unnecessary data. Regarding BC, modules might be using that data? How should it be handled? currently I am only returning what is needed for rendering.

Where was it available ? In the ajax response ?

The fact we are migrating this page means we are making BC anyway. So people will need to update their module 😄 it's a good opportunity to provide them the same capabilities, but done the right way 💪 

I see 2 possible options, there might be more:
1) Provide the same data, maybe different structure (we dont care about BC, if the new structure is better then it's a good thing)
2) Provide a new way to retrieve this data (1 or multiple JSON API endpoints ?) and explain to people they need to pull the data they need => looks better, right ?

But the solution "we do nothing about it" does not look good 😄

**Comment by zuk3975** [comment]
todo: admin_carts_view action has a button "Create an order form this cart". Which redirects to create new order page and loads cart by id.

**Comment by Progi1984** [comment]
Not finished : the check list is not checked ;)

**Comment by sam-pires** [comment]
Reminder: Once the migration part is done, organize a meeting to read all issues, regroup them when possible, and prioritize them.

**Comment by marionf** [comment]
Can we close this issue ?

**Comment by zuk3975** [comment]
> Can we close this issue ?

https://github.com/PrestaShop/PrestaShop/pull/17179
at least this one is blocking

**Comment by colinegin** [comment]
Hello @matks and @zuk3975 can we close this issue now or is there still work to be done on it ?

**Comment by zuk3975** [comment]
> Hello @matks and @zuk3975 can we close this issue now or is there still work to be done on it ?

**There are still some low priority improvements that could be made** (for example carts/orders list could be paginatable, also checklist of this issue is not fully checked yet), but I don't know when its planned to happen. **What was planned for 1.7.7 is finished.**

**Comment by colinegin** [comment]
Wdyt @MatShir @matks ?
I think we should create a new issue if there are some improvements to be done, to be able to close this "master" one.

**Comment by MatShir** [comment]
The issues are already created, the list is in this [epic](https://github.com/PrestaShop/PrestaShop/issues/17598). Since this page is migrated and accessible from the BO, we can close it. We can work on the improvements and bugs directly on the version kanban.
